src/Features/CSharp/Portable/MakeStructFieldsWritable/CSharpMakeStructFieldsWritableDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.CodeQuality;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeAnalysis.CSharp.MakeStructFieldsWritable
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed class CSharpMakeStructFieldsWritableDiagnosticAnalyzer : AbstractCodeQualityDiagnosticAnalyzer
+    {
+        private static readonly DiagnosticDescriptor s_diagnosticDescriptor = CreateDescriptor(
+            IDEDiagnosticIds.MakeStructFieldsWritable,
+            new LocalizableResourceString(nameof(FeaturesResources.Make_readonly_fields_writable), FeaturesResources.ResourceManager, typeof(FeaturesResources)),
+            new LocalizableResourceString(nameof(FeaturesResources.Make_readonly_fields_writable), FeaturesResources.ResourceManager, typeof(FeaturesResources)),
+            isUnneccessary: false);
+
+        public CSharpMakeStructFieldsWritableDiagnosticAnalyzer()
+            : base(ImmutableArray.Create(s_diagnosticDescriptor), GeneratedCodeAnalysisFlags.ReportDiagnostics)
+        {
+        }
+
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+            => DiagnosticAnalyzerCategory.SemanticDocumentAnalysis;
+
+        public override bool OpenFileOnly(Workspace workspace) => true;
```

Comment:
why?  we almost always have this as 'false'.

Reply:
Resolved in [https://github.com/dotnet/roslyn/pull/32063/commits/1d37a4baf4aa558d6bf065d6a019b6893ade0cda](https://github.com/dotnet/roslyn/pull/32063/commits/1d37a4baf4aa558d6bf065d6a019b6893ade0cda)